src/Raven.Server/Documents/DocumentIdWorker.cs

```diff
@@ -178,11 +177,16 @@ private static ByteStringContext.InternalScope UnlikelyGetLowerUnicode(ByteStrin
             if (strLength > MaxIdSize)
                 ThrowDocumentIdTooBig(str);
 
-            int maxStrSize = Encoding.GetMaxByteCount(strLength);
-
             int idSize = JsonParserState.VariableSizeIntSize(strLength);
 
-            int escapePositionsSize = JsonParserState.FindEscapePositionsMaxSize(str);
+            int escapePositionsSize = JsonParserState.FindEscapePositionsMaxSize(str, out var escapedCount);
+
+            /*
+             *  add the size of all control characters
+             *  this is to treat case when we have 2+ control character in a row
+             *  GetMaxByteCount returns smaller size than the actual size with escaped control characters
```

Comment:
Add actual data: string such as `"\0\0"` will be translated to `\u0000\u0000`, etc

Reply:
done